datafusion/optimizer/src/inline_table_scan.rs

```diff
@@ -57,7 +57,7 @@ impl OptimizerRule for InlineTableScan {
                         generate_projection_expr(projection, sub_plan)?;
                     let plan = LogicalPlanBuilder::from(sub_plan.clone())
                         .project(projection_exprs)?
-                        .alias(table_name)?;
+                        .alias(table_name.to_string())?;
```

Comment:
will subquery_alias need similar treatment to have an OwnedTableReference instead of String? or String itself should be just fine?

Reply:
I think it is actually important that the subquery alias is a non parsed string here  (otherwise tests fail). 

This rewrite is effectively inlining the view defition as a named subquery (that doesn't make sense to be in a schema):

```sql
...
FROM <view definition> as "table_name"
```

I will add a comment here.